src/latexify/codegen/function_codegen.py

```diff
@@ -536,6 +536,63 @@ def visit_IfExp(self, node: ast.IfExp) -> str:
         latex += self.visit(node)
         return latex + r", & \mathrm{otherwise} \end{array} \right."
 
+    def visit_Match(self, node: ast.Match) -> str:
+        """Visit a match node"""
+        latex = r"\left\{ \begin{array}{ll} "
+        subject_latex = self.visit(node.subject)
+        for i, match_case in enumerate(node.cases):
+            if len(match_case.body) != 1:
+                raise exceptions.LatexifySyntaxError(
+                    "Multiple statements are not supported in Match nodes."
+                )
+            true_latex = self.visit(match_case.body[0])
+            cond_latex = self.visit(match_case.pattern)
+
+            if i < len(node.cases) - 1:  # no wildcard cases
+                if match_case.guard:
+                    cond_latex = self.visit(match_case.guard)
+                    subject_latex = ""  # getting variable from cond_latex
+                if not cond_latex:
```

Comment:
Don't use generated LaTeX for any conditions. It is not stable and the value may change in the future. You need to always check the original ast for conditioning.

Reply:
Could you please clarify what does "check the original ast for conditioning"? Does it mean that we should parse to get the subject_latex ourselves, rather than getting that from cond_latex?

Reply:
`node` passed to each method has sufficient information to process all codegen rules. The code has to check the inner structure of `node` rather than checking the generated latex.